src/Bridges/Constraint/indicator_sos.jl

```diff
@@ -0,0 +1,185 @@
+    return [(MOI.VectorOfVariables, MOI.SOS1{T}),
+            (MOI.ScalarAffineFunction{T}, BC),
+            ]
+end
+
+function MOIB.added_constraint_types(::Type{<:IndicatorSOS1Bridge{T, S}}) where {T, S <: MOI.AbstractScalarSet}
+    return [(MOI.VectorOfVariables, MOI.SOS1{T}),
+            (MOI.ScalarAffineFunction{T}, S),
+           ]
+end
+
+function concrete_bridge_type(::Type{<:IndicatorSOS1Bridge{T}},
+                              ::Type{<:MOI.AbstractVectorFunction},
+                              ::Type{MOI.IndicatorSet{MOI.ACTIVATE_ON_ONE, S}}) where {T, S<:Union{MOI.LessThan, MOI.GreaterThan}}
+    return IndicatorSOS1Bridge{T, S, MOI.ConstraintIndex{MOI.SingleVariable, S}}
+end
+
+function concrete_bridge_type(::Type{<:IndicatorSOS1Bridge{T}},
+                              ::Type{<:MOI.AbstractVectorFunction},
+                              ::Type{MOI.IndicatorSet{MOI.ACTIVATE_ON_ONE, S}}) where {T, S <: MOI.AbstractScalarSet}
+    return IndicatorSOS1Bridge{T, S, Nothing}
+end
+
+# Attributes, Bridge acting as a model
+
+function MOI.get(::IndicatorSOS1Bridge, ::MOI.NumberOfVariables)
+    return 1
+end
+
+function MOI.get(b::IndicatorSOS1Bridge, ::MOI.ListOfVariableIndices)
+    return [b.w_variable_index]
+end
+
+function MOI.get(::IndicatorSOS1Bridge{T, BC, Nothing}, ::MOI.NumberOfConstraints{MOI.SingleVariable, BC}) where {T, BC}
+    return 0
+end
+
+function MOI.get(::IndicatorSOS1Bridge{T, BC, CI}, ::MOI.NumberOfConstraints{MOI.SingleVariable, BC}) where {T, BC, CI <: MOI.ConstraintIndex{MOI.SingleVariable, BC}}
+    return 1
+end
+
+function MOI.get(::IndicatorSOS1Bridge, ::MOI.NumberOfConstraints{MOI.VectorOfVariables, <:MOI.SOS1})
+    return 1
+end
+
+function MOI.get(::IndicatorSOS1Bridge{T, BC}, ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{T}, BC}) where {T, BC, CI <: MOI.ConstraintIndex{MOI.SingleVariable, BC}}
+    return 1
+end
+
+function MOI.get(b::IndicatorSOS1Bridge{T, BC, CI}, ::MOI.ListOfConstraintIndices{MOI.SingleVariable, BC}) where {T, BC, CI <: MOI.ConstraintIndex}
+    return [b.bound_constraint_index]
+end
+
+function MOI.get(::IndicatorSOS1Bridge{T, BC, Nothing}, ::MOI.ListOfConstraintIndices{MOI.SingleVariable, BC}) where {T, BC}
+    return MOI.ConstraintIndex{MOI.SingleVariable, BC}[]
+end
+
+function MOI.get(b::IndicatorSOS1Bridge{T}, ::MOI.ListOfConstraintIndices{MOI.VectorOfVariables, <:MOI.SOS1}) where {T}
+    return [b.sos_constraint_index]
+end
+
+function MOI.get(b::IndicatorSOS1Bridge{T, BC}, ::MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{T}, BC}) where {T, BC}
+    return [b.linear_constraint_index]
+end
+
+function MOI.supports(
+    ::MOI.ModelLike,
+    ::MOI.ConstraintPrimalStart,
+    ::Type{<:IndicatorSOS1Bridge})
+    return true
+end
+
+function MOI.get(model::MOI.ModelLike, attr::MOI.ConstraintPrimal, bridge::IndicatorSOS1Bridge)
+    if attr.N == 1
```

Comment:
Why do a special case for `N == 1` ? You should not assume anything on what are the different results and their order

Reply:
This is getting the `MOI.ConstraintPrimal` for the function [z, a^T x], if this is one, it means getting the primal of a single variable z, otherwise, it's getting the primal of a Scalar Affine

Reply:
For me this comes from the fact that we are using a vector function, instead of a tuple function: (SingleVariable, AbstractScalarFunction)

Reply:
This should return a vector of 2 values. `N` is simply the index of the result (i.e. usually it is 1). See `MOI.ResultCount`

Reply:
so in that case there are the cases `N = 1` and `N = 2`, we can check one and deduce the other, or throw an error but this makes the functions heavier

Reply:
Not sure to follow, you just need to make sure that the attributes you create in this function have the same result index. You don't assume anything on the nature of the result index.

Reply:
oh ok, this is really not obvious eitheir, getting `attr.N` on a function value implies that you are getting the N-th member of a vector-valued function

Reply:
Indeed, it is planned to be renamed, see https://github.com/JuliaOpt/MathOptInterface.jl/issues/960